src/util/strencodings.cpp

```diff
@@ -46,14 +46,20 @@ bool IsHex(std::string_view str)
     return (str.size() > 0) && (str.size()%2 == 0);
 }
 
-bool IsHexNumber(std::string_view str)
+std::optional<std::string> TrySanitizeHexNumber(std::string_view input, int result_size)
 {
-    if (str.substr(0, 2) == "0x") str.remove_prefix(2);
-    for (char c : str) {
-        if (HexDigit(c) < 0) return false;
+    input = util::RemovePrefixView(input, "0x");
+    if (result_size < 0) result_size = input.size(); // negative result_size disables the size check
```

Comment:
nit: my first reaction was here - "I wonder if there's an off-by-one error here".
If you think it makes sense, consider this alternative:
```suggestion
    if (result_size < 0) result_size = std::numeric_limits<int>::max(); // negative result_size disables the size check
```

Reply:
That wouldn't work. `result_size` is used both to throw if the input is too large, as well as to pad with leading zeroes if it's too short: https://github.com/bitcoin/bitcoin/pull/30569/files#diff-3f688af8f182edecd9c33977b905b3e71dc010574f721fd4e328bdbc7706f574R59

Reply:
K, in that case the test cases check the boundaries correctly already, thanks for considering.